src/tetris/profiler/hardwareController/cameraController.ts

```diff
@@ -0,0 +1,125 @@
+import HardwarePermission from "tetris/profiler/hardwareController/hardwarePermission";
+
+export default class CameraController {
+
+	//region public members
+	public get permissionState(): HardwarePermission {
+		return this._permissionState;
+	}
+	//endregion
+
+	//region public methods
+	public startVideoStream(): Promise<MediaStream> {
+		if (this.permissionState === HardwarePermission.denied) {
+			return Promise.reject(new Error('No Permissions granted'));
+		}
+		return new Promise( (resolve, reject) => {
```

Comment:
The nesting level get's a bit high here. Instead of explicitly creating a promise, I suggest using the `async` and `await` operators.